### ISSUE TITLE
Lawnchair crashes 

### ISSUE THREAD

**Issue by Mr-KIYFD** [Deletescape-Media/Lawnchair#156]
Thanks for the awesome  launcher.
I'm having a slight problem, the launcher works as intended but sometimes after unlocking the screen (I only use fingerprint to unlock, not sure if it's relevant) the launcher crashes and I have to pick a "new" default launcher again.

Phone info : Xiaomi Redmi Note 4
                    MIUI  8.2.1.0 Global
                     Xposed Installed
                     Also got the Launcher3 by Amirzaidi

Crash log 
`                        java.lang.SecurityException: Disallowed call from unknown notification listener: android.service.notification.INotificationListener$Stub$Proxy@d5a1690
	at android.os.Parcel.readException(Parcel.java:1599)
	at android.os.Parcel.readException(Parcel.java:1552)
	at android.app.INotificationManager$Stub$Proxy.getActiveNotificationsFromListener(INotificationManager.java:1046)
	at android.service.notification.NotificationListenerService.getActiveNotifications(NotificationListenerService.java:467)
	at android.service.notification.NotificationListenerService.getActiveNotifications(NotificationListenerService.java:420)
	at ch.deletescape.lawnchair.notification.NotificationListener$C04631.handleMessage(Unknown Source)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loop(Looper.java:207)
	at android.os.HandlerThread.run(HandlerThread.java:61)`

**Comment by deletescape** [comment]
This is really weird and should definitely not happen. Will look into it

**Comment by fonix232** [comment]
By the looks of it, the notification receiver tries to access notifications, but @Mr-KIYFD haven't given Lawnchair access to them, thus an exception is thrown. This seems to be MIUI-specific though.

@Mr-KIYFD can you please try and enable notification access, and see if it still crashes? Just want to verify my theory (can't reproduce it on my S8).

**Comment by Mr-KIYFD** [comment]
I have "notification badges" enabled under notification access, I have had them enabled the whole time. I'm not sure if there are other notification access settings to enable, I tried searching under phone settings and Lawnchair did not appear on the apps listed on the phones notification settings. I hope this is not confusing, I tried uploading sceenshots and it failed

**Comment by edisso10018** [comment]
You must grant notification access for Lawnchair wherever you can find this is in MIUI.

In the 'normal' Android versions it's here:

Settings > Apps > Configure Apps (gear at the top) -> Special Access -> Notification Access

**Comment by Mr-KIYFD** [comment]
I'm closing the issue as it's not crashing anymore.

I'm not sure though as to what fixed it.
The only thing I did was grant notification access for "com.mi.dlabs.vr....." , and the Lawnchair crashed again and became weird (the pill disappeared etc) but no more crashes. I turned off the notification access for com.mi and then uninstalled Lawnchair and reinstalled it and I haven't had a single crash since.
